File: 0768-partition-labels/0768-partition-labels.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        last_idx = defaultdict()
        res=[]
        end=0
        size=0
        for i in range(len(s)):
            last_idx[s[i]]=i
        for i in range(len(s)):
            size+=1
            end=max(end,last_idx[s[i]])
            if i==end:
                res.append(size)
                size=0
        return res
        

File: 0768-partition-labels/0768-partition-labels.cpp
class Solution {
public:
    vector<int> partitionLabels(string s) {
        unordered_map<char, int> last_idx;
        vector<int> res;
        int end = 0, size = 0;
        for(int i = 0; i< s.length();i++){
            last_idx[s[i]]=i;
        }
        for(int i = 0; i< s.length();i++){
            size++;
            end=max(end,last_idx[s[i]]);
            if(i==end){
                res.push_back(size);
                size=0;
            }
        }
        return res;
        
    }
};